packages/data/lib/service/tmdb_api_service.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:data/service/service_payload.dart';
+import 'package:dio/dio.dart';
+
+abstract class TmdbApiService<SP extends ServicePayload> {
```

Comment:
u shouldn't define an abstraction for each ApiServiceImpl. one abstraction - multiple implementation

so, u should have TraktApiService and TmdbApiService classes, which implement ApiService abstract class